packages/wallet/backend/src/paymentPointer/service.ts

```diff
@@ -132,4 +134,84 @@ export class PaymentPointerService implements IPaymentPointerService {
       active: false
     })
   }
+
+  async generateKeyPair(
+    userId: string,
+    paymentPointerId: string
+  ): Promise<{ privateKey: KeyObject; publicKey: KeyObject }> {
+    const paymentPointer = await PaymentPointer.query().findById(
+      paymentPointerId
+    )
+
+    if (!paymentPointer) {
+      throw new NotFound()
+    }
+
+    // Check if the user owns the payment pointer.
+    // This function throws a NotFoundException.
+    await this.deps.accountService.findAccountById(
```

Comment:
move getting the payment pointer and validation of account into payment pointer service